src/server/pmix_server_get.c

```diff
@@ -404,6 +413,29 @@ static pmix_status_t _satisfy_request(pmix_nspace_t *nptr, int rank,
      * having been committed */
     htptr = hts;
     PMIX_CONSTRUCT(&pbkt, pmix_buffer_t);
+
+    /* if they are asking about a rank from an nspace different
+     * from their own, then include a copy of the job-level info */
+    if (NULL != cd &&
+        0 != strncmp(nptr->nspace, cd->peer->info->nptr->nspace, PMIX_MAX_NSLEN)) {
+        cur_rank = PMIX_RANK_WILDCARD;
```

Comment:
@rhc54 Does that mean that we will include the job-level info for each foreign rank?
I mean if some proc is interested in several ranks from the foreign namespace it makes sense to send the job-level data only once.